syntax = "proto3";

message Neighbor {
  string node_id = 1;
  string node_ip = 2;
  int32 control_port = 3;
  int32 data_port = 4;
}

message ControlMessage {
  enum Type {
    REGISTER = 0;
    REGISTER_RESPONSE = 1;
    UPDATE_NEIGHBORS = 2;
    ACK = 3;
  }
  Type type = 1;
  string node_id = 2;
  string node_ip = 3;
  int32 control_port = 4;
  int32 data_port = 5;
  repeated Neighbor neighbors = 6;
}